src/clj_sqs_extended/internal/receive.clj

```diff
@@ -66,31 +65,22 @@
 
 (defn- error-might-be-recovered-by-restarting
   [error]
-  (cond
-    (instance? ReflectiveOperationException error) false
-    (instance? RuntimeException error) false
-    (instance? UnknownHostException error) true
-    (instance? SocketException error) true
-    (instance? UnknownHostException error) true
-    (instance? HttpTimeoutException error) true
-    :else false))
+  (contains? #{UnknownHostException
+               SocketException
+               HttpTimeoutException} (type error)))
```

Comment:
add a unit test for `error-might-be-recovered-by-restarting` to cover all the expected exception types, i.e. the 4-5 types you have originally. this fn is an important part of the error recovery logic, so we should unit test